pilot/pkg/model/service.go

```diff
@@ -56,12 +56,19 @@ type Service struct {
 	// Address specifies the service IPv4 address of the load balancer
 	Address string `json:"address,omitempty"`
 
-	// Protect concurrent ClusterVIPs read/write
+	// Protect concurrent read/write
 	Mutex sync.RWMutex
+
 	// ClusterVIPs specifies the service address of the load balancer
 	// in each of the clusters where the service resides
 	ClusterVIPs map[string]string `json:"cluster-vips,omitempty"`
 
+	// ExternalAddresses is a mapping between a cluster name and the load
+	// balancer address(es) to access the service. Used by the aggregator
+	// to aggregate the Attributes.LoadBalancerAddresses for clusters where
+	// the service resides
+	ExternalAddresses map[string][]string
+
```

Comment:
You cannot change this struct.. All changes to services need to be reflected in the Service Entry as this is what we will stream via Galley as well.

Reply:
I understand but current Pilot impl is creating an aggregated `Service` resource from watched services found by the local/remote k8s controller. No ServiceEntry involved afaik.
Your comment is very true when Galley will serve the remote resources but that's not how it works today.